xdem/spstats.py

```diff
@@ -1003,22 +1009,273 @@ def patches_method(dh : np.ndarray, mask: np.ndarray[bool], gsd : float, area_si
     return df
 
 
-def plot_vgm(df: pd.DataFrame, fit_fun: Callable = None):
-
+def plot_vgm(df: pd.DataFrame, list_fit_fun: Optional[list[Callable[[float],float]]] = None,
+             list_fit_fun_label: Optional[list[str]] = None):
+    """
+    Plot empirical variogram, with optionally one or several model fits
+    :param df: dataframe of empirical variogram
+    :param list_fit_fun: list of function fits
+    :param list_fit_fun_label: list of function fits labels
+    :param
+    :return:
+    """
     fig, ax = plt.subplots(1)
     if np.all(np.isnan(df.exp_sigma)):
         ax.scatter(df.bins, df.exp, label='Empirical variogram', color='blue')
     else:
         ax.errorbar(df.bins, df.exp, yerr=df.exp_sigma, label='Empirical variogram (1-sigma s.d)')
 
-    if fit_fun is not None:
-        x = np.linspace(0, np.max(df.bins), 10000)
-        y = fit_fun(x)
+    if list_fit_fun is not None:
+        for i, fit_fun in enumerate(list_fit_fun):
+            x = np.linspace(0, np.max(df.bins), 10000)
+            y = fit_fun(x)
+
+            if list_fit_fun_label is not None:
+                ax.plot(x, y, linestyle='dashed', label=list_fit_fun_label[i], zorder=30)
+            else:
+                ax.plot(x, y, linestyle='dashed', color='black', zorder=30)
 
-        ax.plot(x, y, linestyle='dashed', color='black', label='Model fit', zorder=30)
+        if list_fit_fun_label is None:
+            ax.plot([],[],linestyle='dashed',color='black',label='Model fit')
 
     ax.set_xlabel('Lag (m)')
     ax.set_ylabel(r'Variance [$\mu$ $\pm \sigma$]')
     ax.legend(loc='best')
-    ax.grid()
-    plt.show()
+
+    return ax
+
+def plot_1d_binning(df: pd.DataFrame, var_name: str, statistic_name: str, label_var: Optional[str] = None,
```

Comment:
Same comment regarding the use of ax, to allow more flexibility.

Reply:
Although this will be a bit more complex since you're using subplots here already. But I wonder if you can generate subplots from an ax instance.

Reply:
Perhaps https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.inset_axes.html?

Reply:
Yes, could work maybe. Just be aware of the warning "This method is experimental as of 3.0, and the API may change."